crates/store/src/state.rs

```diff
@@ -991,67 +1157,70 @@ impl State {
             storage_requests,
         } = detail_request;
 
-        let account_info = self.db.select_historical_account_at(account_id, block_num).await?;
+        if !account_id.is_public() {
+            return Err(DatabaseError::AccountNotPublic(account_id));
+        }
```

Comment:
This should be `account_id.has_public_state()`, otherwise it will error out on network accounts.